crates/composefs/src/repository.rs

```diff
@@ -220,6 +234,10 @@ impl<ObjectID: FsVerityHashValue> Repository<ObjectID> {
 
     /// Basically the same as has_stream() except that it performs expensive verification
     pub fn check_stream(&self, sha256: &Sha256Digest) -> Result<Option<ObjectID>> {
+        if self.insecure {
+            return self.has_stream(sha256);
```

Comment:
This isn't sufficient, I'm afraid.  We want to iterate the full structure to check that it's consistent and matches the given sha256 body digest, even if we don't have fs-verity.